packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -11338,6 +11338,64 @@ describe('NetworkController', () => {
         });
       });
     });
+
+    it('allows calling `getNetworkConfigurationByNetworkClientId` when subscribing to state changes containing new endpoints', async () => {
+      const network = buildCustomNetworkConfiguration({
+        chainId: '0x1' as Hex,
+        name: 'mainnet',
+        nativeCurrency: 'ETH',
+        blockExplorerUrls: [],
+        defaultRpcEndpointIndex: 0,
+        rpcEndpoints: [
+          {
+            type: RpcEndpointType.Custom,
+            url: 'https://test.endpoint/1',
+            networkClientId: 'client1',
+          },
+        ],
+      });
+
+      await withController(
+        {
+          state: {
+            selectedNetworkClientId: 'client1',
+            networkConfigurationsByChainId: { '0x1': network },
+          },
+        },
+        async ({ controller, messenger }) => {
+          // Verify we can call `getNetworkConfigurationByNetworkClientId` with the new endpoint
+          const stateChangePromise = new Promise<void>((resolve, reject) => {
+            messenger.subscribe('NetworkController:stateChange', (state) => {
+              const { networkClientId } =
+                state.networkConfigurationsByChainId['0x1'].rpcEndpoints[1];
+
+              const networkConfiguration =
+                controller.getNetworkConfigurationByNetworkClientId(
+                  networkClientId,
+                );
+
+              networkConfiguration
+                ? resolve()
+                : reject(`networkClient ${networkClientId} not found`);
+            });
```

Comment:
So if we reject, the test would fail?

Could this be refactored to use `expect` assertions?

```ts
let expectedNetworkConfig;
...
messenger.subscribe((...) => {
  ...
  expectedNetworkConfig = networkConfiguration;
})

...

expect(expectedNetworkConfig).toBeDefined()
...
```

If the above snippet fails, lets keep the subscription promise, but resolve with the result? Then once it is awaited we can perform the expect assertions?

Reply:
I'm fine with keeping as is, but it feels a little odd that we don't explicitly assert and rely on promise rejections to fail.

Reply:
yes the promise rejection causes an exception to be thrown which fails the test.  The test failed on the old code and passes on the new code.  I can probably make it an explicit expect instead.

Reply:
https://github.com/MetaMask/core/pull/5122/commits/4b650b82234411df287c2d16ebbea45d8c98eca0